experiment/measurer.py

```diff
@@ -548,22 +559,26 @@ def measure_snapshot_coverage(fuzzer: str, benchmark: str, trial_num: int,
     # Don't keep corpus archives around longer than they need to be.
     os.remove(corpus_archive_dst)
 
-    # Get the coverage of the new corpus units.
+    # Run coverage on the new corpus units.
     snapshot_measurer.run_cov_new_units()
-    all_pcs = snapshot_measurer.merge_new_pcs()
+
+    # Generate profdata and transform it into json form
+    snapshot_measurer.generate_profdata()
+    snapshot_measurer.generate_summary()
+    # Get the coverage of the new corpus units.
+    region_covered = snapshot_measurer.get_current_coverage()
     snapshot = models.Snapshot(time=this_time,
                                trial_id=trial_num,
-                               edges_covered=len(all_pcs))
+                               edges_covered=region_covered)
 
     # Record the new corpus files.
     snapshot_measurer.update_measured_files()
 
     # Archive crashes directory.
     snapshot_measurer.archive_crashes(cycle)
-
-    measuring_time = round(time.time() - measuring_start_time, 2)
-    snapshot_logger.info('Measured cycle: %d in %d seconds.', cycle,
-                         measuring_time)
+    measuring_time = round(time.time() - measuring_start_time, 4)
```

Comment:
Why is this needed at all if I already measure the time on 521?

Reply:
We need to calculate the time spent to measure one cycle. Line 521 is the start time and here we get the end time and then do a subtraction.